lib/include/openamp/rpmsg.h

```diff
@@ -110,6 +110,19 @@ struct rpmsg_device {
 	bool support_ns;
 };
 
+/**
+ * is_rpmsg_ept_ready - check if the rpmsg endpoint ready to send
+ *
+ * @ept: pointer to rpmsg endpoint
+ *
+ * Returns 1 if the rpmsg endpoint has both local addr and destination
+ * addr set, 0 otherwise
+ */
+static inline unsigned int is_rpmsg_ept_ready(struct rpmsg_endpoint *ept)
+{
+	return ept && ept->rdev && ept->dest_addr != RPMSG_ADDR_ANY;
+}
+
```

Comment:
Please could you not move the position of the declaration in file, that a diff show that you only update the return test here.

Reply:
Ok, I will split the movement to another patch

Reply:
For this series , it is sufficient to put the function back to its original place that the diff shows that only the condition as been reworked.
Then LGTM for the rest of the series

Reply:
But, you can view the individual patches in commits tab or git log. I want to keep the patchset as small as possible:
https://github.com/apache/incubator-nuttx/tree/master/openamp

Reply:
Since you are arguing to add poll inside rpmsg_send, I prefer to merge this simple patch first, so the downstream project can redue the patch need to apply even the poll patch can't upstream.

Reply:
Doing this we will simplify your downstream but perhaps impact another one... Not a good practice, if we accept this one, then we would have to accept future patches from another platform that could not match with your expectation.
And  regarding https://github.com/apache/incubator-nuttx/blob/master/openamp/0004-rpmsg-wait-ept-ready-in-rpmsg_send.patch seems to me that it is minor in term of impact in your downstream.

Reply:
But, can you please tell me what will impact by this patch? And it is also good thing to put the simple inline function at the begining like other header files:
https://github.com/OpenAMP/open-amp/blob/master/lib/include/openamp/remoteproc.h#L432-L448
https://github.com/OpenAMP/open-amp/blob/master/lib/include/openamp/rpmsg_virtio.h#L65-L110

Reply:
As explained in my previous comment, this commit seems appears to be for  your convenience only.  Do I accept it to simplify your patches on top of OpenAMP and potentially impact some other platforms which would have to rework their own patch on top of OpenAMP because this one introduce a conflict? I can not find here a good reason to do this as it does not  answer to a real need.